_posts/2022-04-21-ambient-peer-discovery.md

```diff
@@ -0,0 +1,377 @@
+
+Further, this works well with the overload and DoS protection discv5 employs.
+Discv5 limits the amount of nodes registered per topic on a single peer. Further, discv5 enforces a waiting time before nodes can register topics at peers.
+So, for popular topics, a node might fail to register the topic in a close neighbourhood.
+However, because the topic is popular (has a high occurrence percentage $p$), it can still be efficiently discovered.
+
+In the future, we also plan to integrate Waku v2 capability discovery, which will not only allow asking for nodes that support Waku v2,
+but asking for Waku v2 nodes supporting specific Waku v2 protocols like filter or store.
+For the store protocol we envision sub-capabilities reflecting message topics and time frames of messages.
+We will also investigate related security implications.
+
+### Attacks on DHTs
+
+In this post, we only briefly describe common attacks on DHTs.
+These attacks are mainly used for denial of service (DoS),
+but can also used as parts of more sophisticated attacks, e.g. deanonymization attacks.
+A future post on this research log will cover security aspects of ambient peer discovery with a focus on privacy and anonymity.
+
+#### Sybil Attack
+
+The power of an attacker in a DHT is proportional to the number of controlled nodes.
+Controlling nodes comes at a high resource cost and/or requires controlling a botnet via a preliminary attack.
+
+In a Sybil attack, an attacker generates lots of virtual node identities.
+This allows the attacker to control a large portion of the ID space in a DHT at a relatively low cost.
+Sybil attacks are especially powerful when the attacker can freely choose the IDs of generated nodes,
+because this allows positioning at chosen points in the DHT.
+
+Because Sybil attacks amplify the power of many attacks against DHTs,
+making Sybil attacks as difficult as possible is the basis for resilient DHT operation.
+The typical abstract mitigation approach is binding node identities to physical network interfaces.
+To some extend, this can be achieved by introducing IP address based limits.
+Further, generating node IDs can be bound by proof of work (PoW),
+which, however, comes with a set of shortcomings, e.g. relatively high costs on resource restricted devices.
+[The discv5 doc](https://github.com/ethereum/devp2p/blob/6b0abc3d956a626c28dce1307ee9f546db17b6bd/discv5/discv5-rationale.md#sybil-and-eclipse-attacks)
+describes both Sybil and eclipse attacks, as well as concrete mitigation techniques employed by discv5.
+
+
+#### Eclipse Attack
+
+In an eclipse attack, nodes controlled by the attacker poison the routing tables of other nodes in way that parts of the DHT become eclipsed, i.e. invisible.
+When a controlled node is asked for the next step in a path,
+it provides another controlled node as the next step,
+effectively navigating the querying node around or away from certain areas of the DHT.
```

Comment:
One mitigation here is to have alpha > 1 and additionally to follow each concurrent path independently for the lookup. Not sure if you want to mention that.